datafusion/core/src/datasource/listing/table.rs

```diff
@@ -1257,9 +1257,9 @@ mod tests {
 
     #[tokio::test]
     async fn unbounded_csv_table_without_schema() -> Result<()> {
-        let tmp_dir = TempDir::new()?;
```

Comment:
strictly speaking these changes are unecessary -- they are a result of how I found the locations to add `DataFusion::io_error` which was to comment out the `From<std::io::Error>` impl for `DataFusionError`

I can revert them if people prefer, but I actually think having errors fail fast (with `unwrap`) makes them easier to understand and debug when a failure does occur